### PR TITLE
Improve default value for experiments set in RuntimeValueProvider

### PR DESCRIPTION
**Pull request by pabloem**
This fixes a bug where checking for experiments would cause an issue (TypeError).

r: @boyuanzz

### CODE REVIEW

**Comment by boyuanzz** [comment]
LGTM, thanks for fix!

**Comment by pabloem** [comment]
Added a unittest. I'll merge after precommits pass.

**Comment by pabloem** [comment]
I've accidentally created a new branch on the Beam repo for this change. Very sorry about that. I'll delete it after merging, and ensure it doesn't happen again.

File: sdks/python/apache_beam/options/value_provider.py [hunk 1]
```diff
@@ -75,7 +75,7 @@ def __hash__(self):
 
 class RuntimeValueProvider(ValueProvider):
   runtime_options = None
-  experiments = None
+  experiments = set()
```
Comment (aaltay) [inline sdks/python/apache_beam/options/value_provider.py]: Would not it be better to do a `None` check in whereever this is used. If we initialize to set(), we cannot distinguish between an not yet initialized experiments from an actually an empty list.

Comment (boyuanzz) [reply sdks/python/apache_beam/options/value_provider.py]: Is there any difference between an empty list of experiments and no initialization?

**Pull request merged**